src/main/scala/com/typesafe/sbt/web/SbtWeb.scala

```diff
@@ -318,16 +298,34 @@ object SbtWeb extends AutoPlugin {
     nodeModules := nodeModuleGenerators(_.join).map(_.flatten).value
   )
 
+  def webJarsPathPrefix: Def.Initialize[Task[String]] = Def.task {
+    path(s"$WEBJARS_PATH_PREFIX/${moduleName.value}/${version.value}/")
+  }
+
+  def syncExportedAssets(): Def.Initialize[Task[File]] = Def.task {
+    val syncTargetDir = classDirectory.value
+    val syncRequired = TrackLevel.intersection(trackInternalDependencies.value, exportToInternal.value) match {
+      case TrackLevel.TrackAlways =>
+        true
+      case TrackLevel.TrackIfMissing | TrackLevel.NoTracking =>
+        !(syncTargetDir / webJarsPathPrefix.value).exists()
+    }
+    if (syncRequired) {
+      state.value.log.info(s"Exporting Assets for ${moduleName.value}")
+      syncMappings(streams.value.cacheDirectory, exportedMappings.value, syncTargetDir)
```

Comment:
Is `exportedMappings` the task that does actual asset generation work? If so we need to turn this into a dynamic task to make this part lazy.